File: src/main/java/am/petstore/PetStore/pets/controller/CategoryController.java
package am.petstore.PetStore.pets.controller;

import am.petstore.PetStore.pets.service.CategoryService;
import com.fasterxml.jackson.databind.JsonNode;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/category")
public class CategoryController {
    @Autowired
    private CategoryService service;

    @GetMapping("/findAll")
    public ResponseEntity findAll() { return service.findAll(); }

    @PostMapping("/create")
    public ResponseEntity create(@RequestBody JsonNode jsonNode) {
        return service.create(jsonNode);
    }

    @PutMapping("/update")
    public ResponseEntity update(@RequestBody JsonNode jsonNode) {
        return service.update(jsonNode);
    }

    @PostMapping("/delete")
    public ResponseEntity delete(@RequestBody JsonNode jsonNode) {
        return service.delete(jsonNode);
    }
}

File: src/main/java/am/petstore/PetStore/pets/controller/CategoryController.kt
package am.petstore.PetStore.pets.controller


import am.petstore.PetStore.pets.service.CategoryService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import org.springframework.web.multipart.MultipartFile

@RestController
@RequestMapping("/category")
class CategoryController @Autowired constructor(private val service: CategoryService) {

    @GetMapping("/findAll")
    fun findAll(@RequestParam(value = "petId", required = false) petId: String?,
                @RequestParam(value = "categoryId", required = false) categoryId: String?): ResponseEntity<*>? {
        return service.findAll(petId?.toLong(), categoryId?.toLong())
    }

    @PostMapping(value = ["/create"], consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
    fun create(@RequestParam(value = "image", required = false) photo: MultipartFile?,
               @RequestParam(value = "petId", required = false) petId: String?,
               @RequestParam(value = "title", required = false) title: String?,
               @RequestParam(value = "categoryId", required = false) categoryId: String?): ResponseEntity<Map<Any, Any>>? {
        return service.create(photo, title, petId?.toLong(), categoryId?.toLong())
    }

    @PostMapping(value = ["/update/{id}"], consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
    fun update(@PathVariable("id") id: Long?, @RequestParam(value = "image", required = false) photo: MultipartFile?,
               @RequestParam(value = "petId", required = false) petId: String?,
               @RequestParam(value = "title", required = false) title: String?,
               @RequestParam(value = "categoryId", required = false) categoryId: String?): ResponseEntity<Map<Any, Any>>? {
        return service.update(id!!, photo, title, categoryId?.toLong(), petId?.toLong())
    }

    @PutMapping("/delete/{id}")
    fun delete(@PathVariable("id") id: Long?): ResponseEntity<*>? {
        return service.delete(id!!)
    }


}